x-pack/legacy/plugins/infra/public/containers/logs/log_summary/log_summary.test.tsx

```diff
@@ -4,271 +4,176 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
-import { mountHook } from 'test_utils/enzyme_helpers';
+import { renderHook } from '@testing-library/react-hooks';
 
-import { ApolloClientContext } from '../../../utils/apollo_context';
 import { useLogSummary } from './log_summary';
 
-describe('useLogSummary hook', () => {
-  it('provides an empty list of buckets by default', () => {
-    const mockApolloClient = {
-      query: jest.fn(),
-    };
+import { fetchLogSummary } from './api/log_summary';
 
-    const { getLastHookValue } = mountHook(
-      () => useLogSummary('SOURCE_ID', null, 1000, null),
-      createMockApolloProvider(mockApolloClient)
-    );
+// Typescript doesn't know that `fetchLogSummary` is a jest mock.
+// We use a second variable with a type cast to help the compiler further down the line.
+jest.mock('./api/log_summary', () => ({ fetchLogSummary: jest.fn() }));
+const fetchLogSummaryMock = fetchLogSummary as jest.Mock;
```

Comment:
`jest.MockedFunction` is a generic so we could ensure that both the mock implementations as well as the mock manipulation methods are correctly typed:

```suggestion
const fetchLogSummaryMock = fetchLogSummary as jest.MockedFunction<typeof fetchLogSummary>;
```

This also reveals a few incorrect types below.

Reply:
Nice to know :) I fixed already